fsm/join_add_cds.go

```diff
@@ -124,106 +123,106 @@ func JoinAddCdsAction(z *music.Zone) bool {
+		log.Printf("VerifyCdsPublished: %s Using FetchRRset interface\n", zone.Name)
+		err, rrSet := updater.FetchRRset(signer, zone.Name, zone.Name, dns.TypeDNSKEY)
 		if err != nil {
 			log.Printf("Error from updater.FetchRRset: %v\n", err)
 		}
 
 		// Create CDS/CDNSKEY RRsets
-		for _, a := range rrs {
+		for _, a := range rrSet {
 			dnskey, ok := a.(*dns.DNSKEY)
 			if !ok {
 				continue
 			}
 
 			if f := dnskey.Flags & 0x101; f == 257 {
-				cdsmap[dnskey.KeyTag()] = dnskey.ToDS(dns.SHA256).ToCDS()
-				cdnskeymap[dnskey.KeyTag()] = dnskey.ToCDNSKEY()
+				cdsFromKSK[dnskey.KeyTag()] = dnskey.ToDS(dns.SHA256).ToCDS()
+				cdnskeyFromKSK[dnskey.KeyTag()] = dnskey.ToCDNSKEY()
 			}
 		}
 	}
-	keyids := []uint16{}
-	for k, _ := range cdsmap {
+	var keyids []uint16
+	for k, _ := range cdsFromKSK {
 		keyids = append(keyids, k)
 	}
 	log.Printf("Verify Publication of CDS: there are KSKs at the signers with the following keytags: %v\n", keyids)
 
-	// Check against published CDS/CDNSKEY RRsets.
-	for _, signer := range z.SGroup.SignerMap {
+	// Compare DNSKEYs to published CDS/CDNSKEY RRsets.
+	for _, signer := range zone.SGroup.SignerMap {
 		updater := music.GetUpdater(signer.Method)
-		err, cdsrrs := updater.FetchRRset(signer, z.Name, z.Name, dns.TypeCDS)
+		err, cdsRRset := updater.FetchRRset(signer, zone.Name, zone.Name, dns.TypeCDS)
 		if err != nil {
-			err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CDS RRset from %s: %v",
+			err, _ = zone.SetStopReason(fmt.Sprintf("Unable to fetch CDS RRset from %s: %v",
 				signer.Name, err))
 			return false
 		}
-		err, cdnskeyrrs := updater.FetchRRset(signer, z.Name, z.Name, dns.TypeCDNSKEY)
+		err, cdnskeyRRset := updater.FetchRRset(signer, zone.Name, zone.Name, dns.TypeCDNSKEY)
 		if err != nil {
-			err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CDNSKEY RRset from %s: %v",
+			err, _ = zone.SetStopReason(fmt.Sprintf("Unable to fetch CDNSKEY RRset from %s: %v",
 				signer.Name, err))
 			return false
 		}
 
-		for _, cdsrr := range cdsrrs { // check all CDS RRs from this signer
-			cds, ok := cdsrr.(*dns.CDS)
+		for _, rr := range cdsRRset {
+			cdsRR, ok := rr.(*dns.CDS)
 			if !ok {
 				continue
 			}
-			cdsmap2[cds.KeyTag] = cds // put the CDS into cdsmap2
-			if _, exist := cdsmap[cds.KeyTag]; !exist {
-				err, _ = z.SetStopReason(fmt.Sprintf("CDS RR with keyid=%d published by signer %s should not exist", cds.KeyTag, signer.Name))
+			cdsFromSigners[cdsRR.KeyTag] = cdsRR
+			if _, exist := cdsFromKSK[cdsRR.KeyTag]; !exist {
+				err, _ = zone.SetStopReason(fmt.Sprintf("CDS RR with keyid=%d published by signer %s should not exist", cdsRR.KeyTag, signer.Name))
 				return false
 			}
 		}
-		for _, revcds := range cdsmap {
-			if _, exist := cdsmap2[revcds.KeyTag]; !exist {
-				err, _ = z.SetStopReason(fmt.Sprintf("CDS RR with keyid=%d should be published by %s, but is not",
-					revcds.KeyTag, signer.Name))
+		for _, cdsRR := range cdsFromKSK {
```

Comment:
Here cdsRR is a *dns.RR.

I don't think this improves readability.

Reply:
As far as I can tell it is a `*dns.CDS`

Reply:
You're right, I'm wrong. Sorry.